x-pack/plugins/logstash/server/routes/pipeline/save.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { ICustomClusterClient, IRouter } from 'src/core/server';
+
+import { INDEX_NAMES } from '../../../common/constants';
+import { Pipeline } from '../../models/pipeline';
+import { licenseCheckerRouteHandlerWrapper } from '../../../../licensing/server';
+import { SecurityPluginSetup } from '../../../../security/server';
+import { checkLicense } from '../../lib/check_license';
+
+export function registerPipelineSaveRoute(
+  router: IRouter,
+  esClient: ICustomClusterClient,
+  security?: SecurityPluginSetup
+) {
+  router.put(
+    {
+      path: '/api/logstash/pipeline/{id}',
+      validate: {
+        params: schema.object({
+          id: schema.string(),
+        }),
+        body: schema.object({
+          id: schema.string(),
+          description: schema.string(),
+          pipeline: schema.string(),
+          username: schema.string(),
+          settings: schema.maybe(schema.object({}, { unknowns: 'allow' })),
```

Comment:
This can be replaced with `schema.maybe(schema.recordOf(schema.string(), schema.any()))` ?

Reply:
I'd keep it as is. All `settings` keys are known and should be declared by someone familiar with this domain later.

Reply:
SGTM